packages/calcite-components/src/components/list/list.e2e.ts

```diff
@@ -252,6 +252,69 @@ describe("calcite-list", () => {
     expect(visibleItems.map((item) => item.id)).toEqual(["label-match", "description-match", "value-match"]);
   });
 
+  it("should support shift click to select multiple items", async () => {
+    const clickItemContent = (item: HTMLCalciteListItemElement, selector: string) => {
+      item.shadowRoot.querySelector(selector).dispatchEvent(new MouseEvent("click", { bubbles: true, shiftKey: true }));
+    };
+
+    const page = await newE2EPage();
+    await page.setContent(html`<calcite-list selection-mode="multiple">
+      <calcite-list-item id="item-1" label="hello" description="world"></calcite-list-item>
+      <calcite-list-item id="item-2" label="hello 2" description="world 2"></calcite-list-item>
+      <calcite-list-item id="item-3" label="hello 3" description="world 3"></calcite-list-item>
+      <calcite-list-item id="item-4" label="hello 4" description="world 4"></calcite-list-item>
+    </calcite-list>`);
+    await page.waitForChanges();
+    await page.waitForTimeout(listDebounceTimeout);
+
+    const list = await page.find("calcite-list");
+    const items = await page.findAll("calcite-list-item");
+
+    items.forEach(async (item) => expect(await item.getProperty("selected")).toBe(false));
+
+    const eventSpy = await list.spyOnEvent("calciteListChange");
+
+    await items[0].click();
+
+    await page.waitForChanges();
+    await page.waitForTimeout(listDebounceTimeout);
+    expect(eventSpy).toHaveReceivedEventTimes(1);
+    expect(await list.getProperty("selectedItems")).toHaveLength(1);
+
+    expect(await items[0].getProperty("selected")).toBe(true);
+    expect(await items[1].getProperty("selected")).toBe(false);
+    expect(await items[2].getProperty("selected")).toBe(false);
+    expect(await items[3].getProperty("selected")).toBe(false);
+
+    await page.$eval("#item-4", clickItemContent, `.${CSS.contentContainer}`);
+    await page.waitForChanges();
+    await page.waitForTimeout(listDebounceTimeout);
+    expect(eventSpy).toHaveReceivedEventTimes(2);
+    expect(await list.getProperty("selectedItems")).toHaveLength(4);
+
+    items.forEach(async (item) => expect(await item.getProperty("selected")).toBe(true));
```

Comment:
`forEach` expects a synchronous function so using async/await doesn't always do what you expect. 

Can you change this to a `for...of` loop? Also applies to the two other instances in this `it` block.

Reply:
Ill just get rid of the loop.